internals/secrethub/credential_list.go

```diff
@@ -0,0 +1,76 @@
+package secrethub
+
+import (
+	"fmt"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+	"github.com/secrethub/secrethub-cli/internals/secrethub/command"
+)
+
+// CredentialListCommand creates a backup code to restore a credential from a code.
+type CredentialListCommand struct {
+	io            ui.IO
+	newClient     newClientFunc
+	useTimestamps bool
+}
+
+// NewAccountInitCommand creates a new CredentialListCommand.
+func NewCredentialListCommand(io ui.IO, newClient newClientFunc) *CredentialListCommand {
+	return &CredentialListCommand{
+		io:        io,
+		newClient: newClient,
+	}
+}
+
+// Register registers the command, arguments and flags on the provided Registerer.
+func (cmd *CredentialListCommand) Register(r command.Registerer) {
+	clause := r.Command("ls", "List all your credentials.")
+	clause.Alias("list")
+
+	registerTimestampFlag(clause).BoolVar(&cmd.useTimestamps)
+
+	command.BindAction(clause, cmd.Run)
+}
+
+// Run lists all the currently authenticated account's credentials.
+func (cmd *CredentialListCommand) Run() error {
+	client, err := cmd.newClient()
+	if err != nil {
+		return err
+	}
+
+	credentials, err := client.Credentials().ListMine()
+	if err != nil {
+		return err
+	}
+
+	timeFormatter := NewTimeFormatter(cmd.useTimestamps)
+
+	w := tabwriter.NewWriter(cmd.io.Stdout(), 0, 2, 2, ' ', 0)
+	fmt.Fprintln(w,
```

Comment:
Could be useful to have this in a helper function that just renders a table in a consistent way.

Reply:
If we pick this up, also check out the audit and service ls tables which take an object (`api.AuditEvent`, `api.Service`) and turn it into a table row. This is used to differentiate the table output (e.g. for secret/repo audit events and for key/aws services).
The table writer solution we create should work together with a table row creator solution.

Shall we pick this up in a separate PR? We use the tables in a lot of places, so that would be a bit much to pick up here.

Reply:
Agreed and agreed.